corehq/apps/domain/forms.py

```diff
@@ -464,6 +464,11 @@ class DomainMetadataForm(DomainGlobalSettingsForm, SnapshotSettingsMixin):
         required=False,
         choices=[]
     )
+    sms_mobile_worker_registration_enabled = BooleanField(
+        label=_("SMS Mobile Worker Registration"),
+        required=False,
+        help_text=_("Turn this on to allow mobile worker registration through SMS")
+    )
```

Comment:
We shouldn't be adding this option to the project settings page, but instead to the sms settings page. We can model the field after this one: https://github.com/dimagi/commcare-hq/blob/register-mobile-worker-sms/corehq/apps/sms/forms.py#L339